tests/integ/test_dataframe.py

```diff
@@ -1699,6 +1699,62 @@ def test_dataframe_duplicated_column_names(session):
     assert "duplicate column name 'A'" in str(ex_info)
 
 
+def test_case_insensitive_collect(session):
+    df = session.create_dataframe(
+        [["Gordon", 153]], schema=["firstname", "matches-won"]
+    )
+    df_quote = session.create_dataframe(
+        [["Gordon", 153]], schema=["'quotedName'", "quoted-won"]
+    )
+
+    # tests for sync collect
+    row = df.collect(case_insensitive=True)[0]
+    assert row.firstName == "Gordon"
+    assert row.FIRSTNAME == "Gordon"
+    assert row.FiRstNamE == "Gordon"
+    assert row["firstname"] == "Gordon"
+    assert row["FIRSTNAME"] == "Gordon"
+    assert row["FirstName"] == "Gordon"
+
+    assert row.matches_won == 153
+    assert row.MATCHES_WON == 153
+    assert row.MaTchEs_WoN == 153
+    assert row["matches_won"] == 153
+    assert row["Matches-Won"] == 153
+    assert row["MATCHES_WON"] == 153
```

Comment:
how does the team feel about this change? I am not only using lower casing to implement this but I am also replacing `-` with `_`

Reply:
Does case insensitive implies `"-" == "_"`? If not then I think we probably should not do this.

Reply:
I earlier thought it could be a good idea but now I'm pretty sure it's gonna break some users and make them mad.